services/web/server/src/simcore_service_webserver/garbage_collector_tasks_users.py

```diff
@@ -0,0 +1,109 @@
+"""
+    Scheduled tasks addressing users
+
+"""
+
+import asyncio
+import logging
+from typing import AsyncIterator, Callable
+
+from aiohttp import web
+from aiopg.sa.engine import Engine
+from models_library.users import UserID
+from servicelib.logging_utils import log_context
+from tenacity import retry
+from tenacity.before_sleep import before_sleep_log
+from tenacity.wait import wait_exponential
+
+from ._constants import APP_DB_ENGINE_KEY
+from .login.utils import notify_user_logout
+from .security_api import clean_auth_policy_cache
+from .users_db import update_expired_users
+
+logger = logging.getLogger(__name__)
+
+CleanupContextFunc = Callable[[web.Application], AsyncIterator[None]]
+
+_SEC = 1
+
+_PERIODIC_TASK_NAME = f"{__name__}.update_expired_users_periodically"
+_APP_TASK_KEY = f"{_PERIODIC_TASK_NAME}.task"
+
+
+async def notify_user_logout_all_sessions(
+    app: web.Application, user_id: UserID
+) -> None:
+    #  NOTE kept here for convenience
+    with log_context(
+        logger,
+        logging.INFO,
+        "Forcing logout of %s from all sessions",
+        f"{user_id=}",
+    ):
+        try:
+            await notify_user_logout(app, user_id, client_session_id=None)
+        except Exception:  # pylint: disable=broad-except
+            logger.warning(
+                "Ignored error while notifying logout for %s",
+                f"{user_id=}",
+                exec_info=True,
+            )
+
+
+@retry(
+    wait=wait_exponential(min=5 * _SEC),
+    before_sleep=before_sleep_log(logger, logging.WARNING),
+)
+async def _update_expired_users_periodically(app: web.Application, wait_s: float):
+    """Periodically check expiration dates and updates user status
+
+    It is resilient, i.e. if update goes wrong, it waits a bit and retries
+    """
+    engine: Engine = app[APP_DB_ENGINE_KEY]
+    assert engine  # nosec
+
+    while True:
```

Comment:
minor: you can also do while await asyncio.sleep(X, result=True):

Reply:
i like the technical aspect of it but it changes the logic to wait first, and I rather have do and then wait, i.e. wait last. This way we can restart the service and be sure that running is the first thing it does